syft/pkg/cataloger/cataloger.go

```diff
@@ -43,6 +44,7 @@ func ImageCatalogers(cfg Config) []Cataloger {
 		java.NewJavaCataloger(cfg.Java()),
 		apkdb.NewApkdbCataloger(),
 		golang.NewGoModuleBinaryCataloger(),
+		dartlang.NewPubspecLockCataloger(),
```

Comment:
question: is a pubspec lock file a solid indication of the package being positively installed? or is the presence of this file not necessarily indicative of it being installed (say if it's in a source repo)?

(We reserve the set of image catalogers by default to only be the set of catalogers that raise up installed packages)

Reply:
I would say it's akin to a go.sum or pip.lock. My current use case is to inspect artifacts generated by my companies internal build system and hope to use syft to identify that a pubspec.lock file was generated. We happen to use Docker to build said artifacts I suppose scanning a container is not required for our usecase.

Reply:
I will remove it for now, we can explore it in the future if we need to.